neqo-http3/src/features/extended_connect/tests/webtransport/mod.rs

```diff
@@ -97,7 +97,7 @@ fn connect_with(client: &mut Http3Client, server: &mut Http3Server) {
     let out = client.process(out.dgram(), now());
     let out = server.process(out.dgram(), now());
     let out = client.process(out.dgram(), now());
-    let _ = server.process(out.dgram(), now());
+    std::mem::drop(server.process(out.dgram(), now()));
```

Comment:
Although the build was successful with my initial commit. The CI reported failure due to clippy warnings.
Hence I corrected all the clippy warnings in the second commit. The list of all the clippy warnings can be seen [here](https://github.com/mozilla/neqo/actions/runs/3549518560/jobs/5962000160)

The above change was intended because of the following error:
```
 error: non-binding `let` on a type that implements `Drop`
   --> neqo-http3/src/features/extended_connect/tests/webtransport/mod.rs:100:5
    |
100 |     let _ = server.process(out.dgram(), now());
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: consider using an underscore-prefixed named binding or dropping explicitly with `std::mem::drop`
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#let_underscore_drop

```

Reply:
Thanks for the explanation.